.readthedocs.yml

```diff
@@ -4,6 +4,14 @@ build:
   os: ubuntu-22.04
   tools:
     python: '3.10'
+  commands:
+    - python -mvirtualenv $READTHEDOCS_VIRTUALENV_PATH
+    - curl -fsSL https://d2lang.com/install.sh | sh -s --
+    - python -m pip install --upgrade --no-cache-dir pip setuptools
+    - python -m pip install --upgrade --no-cache-dir mkdocs
+    - python -m pip install --exists-action=w --no-cache-dir -r docs/requirements.txt
+    - cat mkdocs.yml
+    - PATH=$HOME/.local/bin:$PATH python -m mkdocs build --clean --site-dir $READTHEDOCS_OUTPUT/html --config-file mkdocs.yml
```

Comment:
Why is this necessary?

Reply:
The d2 dependency, and the fact that d2 is not in the apt repos yet.

Reply:
Can we use just `pre_build` (https://docs.readthedocs.io/en/stable/build-customization.html) so that we can still take advantage of the default build steps?

Reply:
Great idea, I didn't see that.

Reply:
Looks like this was changed in 86c185c.